src/averageAges.js

```diff
@@ -1,62 +1,56 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function getAge(deathYear, birthYear) {
+  return deathYear - birthYear;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter(person => person.sex === 'm');
+
+  if (century) {
+    men = men.filter(man => Math.ceil(man.died / 100) === century);
+
+    return men.reduce((acc, man) =>
+      getAge(man.died, man.born) + acc, 0) / men.length;
+  } else {
+    return men.reduce((acc, man) =>
+      getAge(man.died, man.born) + acc, 0) / men.length;
+  }
```

Comment:
You could get rid of: 
- `if-else` by using logical operator inside `filter` to validate two conditions at a time 
- duplicating `reduce` by moving these operation off the main function to a separate one